src/control/server/ctl_storage_rpc.go

```diff
@@ -506,13 +506,39 @@ func (c *ControlService) adjustScmSize(resp *ctlpb.ScanScmResp) {
 	}
 }
 
+func checkEnginesReady(instances []Engine) error {
+	for _, inst := range instances {
+		if !inst.IsReady() {
+			var err error = FaultDataPlaneNotStarted
+			if inst.IsStarted() {
+				err = errInstanceNotReady
+			}
+
+			return errors.Wrapf(err, "instance %d", inst.Index())
+		}
+	}
+
+	return nil
+}
+
 // StorageScan discovers non-volatile storage hardware on node.
 func (c *ControlService) StorageScan(ctx context.Context, req *ctlpb.StorageScanReq) (*ctlpb.StorageScanResp, error) {
 	if req == nil {
 		return nil, errors.New("nil request")
 	}
 	resp := new(ctlpb.StorageScanResp)
 
+	// In the case that usage stats are being requested, relevant flags for both SCM and NVMe
+	// will be set and so fail if engines are not ready for comms. This restriction should not
+	// be applied if only the Meta flag is set in the NVMe component of the request to continue
+	// to support off-line storage scan functionality which uses cached stats (e.g. dmg storage
+	// scan --nvme-meta).
+	if req.Scm.Usage && req.Nvme.Meta {
```

Comment:
Based on the comment, I'm wondering, is checking `req.Nvme.Meta` necessary at all? Is it possible to query only SCM usage without NVMe?

Reply:
it's already a bit of a hack extrapolating intent from these derived flags, relying on just the Scm.Usage flag seems like a worse hack. Seems like depending on both would prevent strange behaviour if someone introduced Scm only usage scan request.

Reply:
Hm, okay. I would have thought any request for the usage stats could result in this issue.

Maybe it would be best to make this check at the point where we actually need something from the running engines? We wouldn't abort as quickly, but maybe it would feel less hacky.

Reply:
We do make "engine ready" checks elsewhere further down the call stack but the reason I want to do it here is to avoid performing an SCM usage scan before realising dRPC isn't ready. I think it's best to check that first and return early.

Reply:
All right. I do agree it's good to pre-empt it if we have good signals, but my issue is just what you mentioned above: "extrapolating intent from these derived flags." I'm okay with landing as-is for now, but maybe we should revisit the messages we use for StorageScan if they're not describing the user's intent clearly enough in this case.